core/src/main/scala/org/apache/spark/storage/ShuffleBlockFetcherIterator.scala

```diff
@@ -212,6 +214,9 @@ final class ShuffleBlockFetcherIterator(
         logWarning("Failed to cleanup shuffle fetch temp file " + file.path())
       }
     }
+    // Unregister completion callbacks to avoid memory leak for ShuffleBlockFetcherIterator
+    // contains lots of metadata about MapStatus
+    context.removeTaskCompletionListener(cleanupListener)
```

Comment:
```
[error]  * abstract method removeTaskCompletionListener(org.apache.spark.util.TaskCompletionListener)org.apache.spark.TaskContext in class org.apache.spark.TaskContext is present only in current version
[error]    filter with: ProblemFilters.exclude[ReversedMissingMethodProblem]("org.apache.spark.TaskContext.removeTaskCompletionListener")
```

If `TaskContext` is meant to be an API, old classes that implemented `TaskContext` will fail here in newer Spark since that doesn't implement `removeTaskCompletionListener`. Looks it's not meant to be an API tho.

Reply:
@HyukjinKwon got it, Thanks!